frontend/src/concepts/pipelines/content/artifacts/ArtifactUriLink.tsx

```diff
@@ -11,58 +11,99 @@ interface ArtifactUriLinkProps {
 }
 
 export const ArtifactUriLink: React.FC<ArtifactUriLinkProps> = ({ artifact }) => {
-  const isClassificationMetrics = artifact.getType() === ArtifactType.CLASSIFICATION_METRICS;
+  const isMetrics =
+    artifact.getType() === ArtifactType.CLASSIFICATION_METRICS ||
+    artifact.getType() === ArtifactType.METRICS ||
+    artifact.getType() === ArtifactType.SLICED_CLASSIFICATION_METRICS;
+
+  const isDownloadableOnly = artifact.getType() === ArtifactType.MODEL;
+  const { getStorageObjectDownloadUrl, getStorageObjectRenderUrl } = useArtifactStorage();
   const [loading, setLoading] = React.useState<boolean>(false);
 
-  const { getStorageObjectUrl } = useArtifactStorage();
+  const handleDownload = () => {
+    getStorageObjectDownloadUrl(artifact)
+      .then((url) => {
+        if (url) {
+          triggerFileDownload('', url);
+        }
+      })
+      .catch((error) => {
+        // eslint-disable-next-line no-console
+        console.error('Failed to get the storage URL:', error);
+      })
+      .finally(() => {
+        setLoading(false);
+      });
+  };
 
-  const handleOnClick = async () => {
-    const url = await getStorageObjectUrl(artifact);
-    return url;
+  const handlePreview = () => {
+    getStorageObjectRenderUrl(artifact)
+      .then((url) => {
+        if (url) {
+          window.open(url, '_blank');
+        }
+      })
+      .catch((error) => {
+        // eslint-disable-next-line no-console
+        console.error('Failed to get the storage URL:', error);
+      })
+      .finally(() => {
+        setLoading(false);
+      });
   };
 
   const uri = artifact.getUri();
 
-  if (isClassificationMetrics) {
-    return uri;
+  // If the artifact is a metrics type, there's no need to show the uri
+  // because it's not linked to anything
+  if (isMetrics) {
+    return '-';
   }
 
   return (
     <Flex
       alignItems={{ default: 'alignItemsCenter' }}
       flexWrap={{ default: 'nowrap' }}
-      spaceItems={{ default: 'spaceItemsNone' }}
+      spaceItems={{ default: 'spaceItemsSm' }}
     >
       <FlexItem>
         <Button
           isLoading={loading}
           variant="link"
           isInline
-          icon={<ExternalLinkAltIcon />}
+          icon={isDownloadableOnly ? <DownloadIcon /> : <ExternalLinkAltIcon />}
```

Comment:
why make this button be one or the other. can you just do 
```tsx
<ButtonForDownload />
{!downloadOnly && <ButtonForPreview />}
```

Reply:
The icon links to the text, and they are in the same button, so the primary action for different types is different, Thus, we need to do it in this way.